aws/resource_aws_workspaces_directory.go

```diff
@@ -144,17 +144,30 @@ func resourceAwsWorkspacesDirectoryCreate(d *schema.ResourceData, meta interface
 	}
 	log.Printf("[DEBUG] WorkSpaces Directory %q is registered", directoryId)
 
-	log.Printf("[DEBUG] Modifying WorkSpaces Directory %q self-service permissions...", directoryId)
 	if v, ok := d.GetOk("self_service_permissions"); ok {
+		log.Printf("[DEBUG] Modifying WorkSpaces Directory %q self-service permissions...", directoryId)
 		_, err := conn.ModifySelfservicePermissions(&workspaces.ModifySelfservicePermissionsInput{
 			ResourceId:             aws.String(directoryId),
 			SelfservicePermissions: expandSelfServicePermissions(v.([]interface{})),
 		})
 		if err != nil {
 			return fmt.Errorf("error setting self service permissions: %s", err)
 		}
+		log.Printf("[DEBUG] WorkSpaces Directory %q self-service permissions are set", directoryId)
+	}
+
+	if v, ok := d.GetOk("ip_group_ids"); ok {
+		ipGroupIds := v.(*schema.Set).List()
+		log.Printf("[DEBUG] Associate WorkSpaces Directory %q with IP Groups %s...", directoryId, ipGroupIds)
+		_, err := conn.AssociateIpGroups(&workspaces.AssociateIpGroupsInput{
+			DirectoryId: aws.String(directoryId),
+			GroupIds:    expandStringList(ipGroupIds),
```

Comment:
expandStringSet and there wont be a need to call .List()

Reply:
Awesome 🙂